app/services/user/create.rb

```diff
@@ -0,0 +1,20 @@
+class User::Create < ApplicationService
+  def initialize(params)
+    @user = params[:user]
+  end
+
+  def call
+    create_profile(@user) if save_user(@user)
+    @user
+  end
+
+  protected
+
+  def save_user(user)
+    user.save
+  end
+
+  def create_profile(user)
```

Comment:
You don't need to pass an argument to methods as you have `@user` instance variable which is accessible from any instance method. 

To make it nicer we can create a getter `attr_reader :user`